File: test/fixtures/basic/android/src/main/java/com/example/basic/BasicPlugin.java
package com.example.basic;

import com.getcapacitor.JSObject;
import com.getcapacitor.Plugin;
import com.getcapacitor.PluginCall;
import com.getcapacitor.PluginMethod;
import com.getcapacitor.annotation.CapacitorPlugin;

@CapacitorPlugin(name = "Basic")
public class BasicPlugin extends Plugin {

    private String mode = "fast";

    @PluginMethod
    public void getStatus(PluginCall call) {
        JSObject ret = new JSObject();
        ret.put("enabled", true);
        ret.put("mode", mode);
        ret.put("label", "ok");
        call.resolve(ret);
    }

    @PluginMethod
    public void refresh(PluginCall call) {
        JSObject data = new JSObject();
        data.put("mode", mode);
        notifyListeners("statusChanged", data);
        call.resolve();
    }
}

File: test/fixtures/basic/ios/Sources/BasicPlugin.swift
import Foundation
import Capacitor

@objc(BasicPlugin)
public class BasicPlugin: CAPPlugin, CAPBridgedPlugin {
    public let identifier = "BasicPlugin"
    public let jsName = "Basic"
    public let pluginMethods: [CAPPluginMethod] = [
        CAPPluginMethod(name: "getStatus", returnType: CAPPluginReturnPromise),
        CAPPluginMethod(name: "refresh", returnType: CAPPluginReturnPromise)
    ]

    private var mode = "fast"

    @objc func getStatus(_ call: CAPPluginCall) {
        call.resolve([
            "enabled": true,
            "mode": mode,
            "label": "ok",
            "note": NSNull(),
            "count": 3
        ])
    }

    @objc func refresh(_ call: CAPPluginCall) {
        notifyListeners("statusChanged", data: ["mode": mode])
        call.resolve()
    }
}
